comp/otelcol/otlp/components/exporter/serializerexporter/factory.go

```diff
@@ -156,10 +156,21 @@ func (f *factory) Reporter(params exp.Settings, forwarder defaultforwarder.Forwa
 	return f.reporter, reporterErr
 }
 
+// checkAndCastConfig checks the configuration type and its warnings, and casts it to
+// the Datadog Config struct.
+func checkAndCastConfig(c component.Config, logger *zap.Logger) *ExporterConfig {
+	cfg, ok := c.(*ExporterConfig)
+	if !ok {
+		panic("programming error: config structure is not of type *ExporterConfig")
```

Comment:
This is fine, but I normally prefer to avoid panics... If this is a generalized pattern, then it's OK, but I'm wondering if it wouldn't be better to return `nil`, or change the function to also return an `error`, and then return the error with `createMetricExporter` (which I think would have the same effect and prevent the process from running as well)?